File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int curr = 0;
        
        for(int k : section){
            if (k > curr){
                curr = k + m - 1;
                answer ++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스_코틀린/level1/덧칠하기.kt
package 프로그래머스_코틀린.level1

class Solution46 {
    fun solution(n: Int, m: Int, section: IntArray): Int {
        var answer: Int = 0
        var sectionList = section.toMutableList()
        while (sectionList.size > 0){
            var start = sectionList.first()
            var last = start + m
            sectionList.removeIf{ it >= start && it < last}
            answer += 1
        }
        return answer
    }
}